conans/client/build/compiler_flags.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+from conans.tools import unix_path
+
+
+def rpath_flags(os_build, compiler, lib_paths):
+    if not os_build:
+        return []
+    if compiler in ("clang", "apple-clang", "gcc"):
+        rpath_separator = "," if os_build == "Macos" else "="
+        return ['-Wl,-rpath%s"%s"' % (rpath_separator, x.replace("\\", "/"))
+                for x in lib_paths if x]
+    return []
+
+
+def architecture_flag(compiler, arch):
+    """
+    returns flags specific to the target architecture and compiler
+    """
+    if not compiler or not arch:
+        return ""
+
+    if str(compiler) in ['gcc', 'apple-clang', 'clang', 'sun-cc']:
+        if str(arch) in ['x86_64', 'sparcv9']:
+            return '-m64'
+        elif str(arch) in ['x86', 'sparc']:
+            return '-m32'
+    return ""
+
+
+def libcxx_define(compiler, libcxx):
+
```

Comment:
architectre_flag checks for None value of compiler and arch
seems like this function should check as well, same to others (e.g. pic_flag)